src/org/exist/xquery/XQueryContext.java

```diff
@@ -2045,9 +2030,10 @@ public boolean isVarDeclared( QName qname )
      * 
      * @return currently visible local variables as a stack
      */
-    public List<Variable> getLocalStack() {
-    	final List<Variable> variables = new ArrayList<Variable>(10);
-    	
+    public List<ClosureVariable> getLocalStack() {
+
+        final List<ClosureVariable> closure = new ArrayList<>(6);
```

Comment:
This is an arbitrary choice. Depends on how many let/for or function parameters you expect on average.